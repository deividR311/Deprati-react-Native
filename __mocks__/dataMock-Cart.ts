export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
