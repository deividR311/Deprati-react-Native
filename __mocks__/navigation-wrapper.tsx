import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
