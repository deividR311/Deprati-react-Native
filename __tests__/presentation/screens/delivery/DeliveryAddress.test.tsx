import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import DeliveryAddress from '../../../../src/presentation/screens/Delivery/DeliveryAddress/DeliveryAddress';

describe('DeliveryAddress', () => {
  const mockInitParams = {
    dataCart: {},
    setTitleContinueButton: jest.fn(),
    enableContinueButton: jest.fn(),
    showActivityIndicatorContinueButton: jest.fn(),
    showTotalsBand: jest.fn(),
    setCurrentStep: jest.fn()
  };

  it('renders correctly in Snapshot', () => {
    const screenTest = render(
      <NavigationWrapper
        screen={DeliveryAddress}
        initialParams={mockInitParams}
      />
    );
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });
});
