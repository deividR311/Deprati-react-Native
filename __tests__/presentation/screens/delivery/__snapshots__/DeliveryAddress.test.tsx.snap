// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryAddress renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "paddingBottom": 200,
              }
            }
            showsVerticalScrollIndicator={false}
          >
            <View>
              <View
                accessibilityRole="radiogroup"
              >
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#f6f6f6",
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "disabled": false,
                          "expanded": true,
                        }
                      }
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "padding": 8,
                            },
                            Array [
                              Object {
                                "backgroundColor": "#ffffff",
                                "borderBottomWidth": 0.5,
                                "borderColor": "#979797",
                                "borderTopWidth": 0.5,
                                "height": 50,
                                "justifyContent": "center",
                                "width": "100%",
                              },
                            ],
                          ],
                        ]
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "margin": 8,
                              },
                              Object {
                                "flex": 1,
                                "justifyContent": "center",
                              },
                            ]
                          }
                        >
                          <Text
                            numberOfLines={1}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontWeight": "400",
                                },
                                Object {
                                  "textAlign": "left",
                                },
                                Array [
                                  Object {
                                    "fontSize": 16,
                                  },
                                  Object {
                                    "color": "#6200ee",
                                  },
                                  Object {
                                    "color": "#263238",
                                    "fontFamily": "Roboto-Regular",
                                    "fontSize": 14,
                                    "fontWeight": "600",
                                    "lineHeight": 20,
                                  },
                                ],
                              ]
                            }
                          >
                            Enviar a mi dirección principal
                          </Text>
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "margin": 8,
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            pointerEvents="box-none"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "height": 40,
                                  "justifyContent": "center",
                                  "margin": 8,
                                  "width": 40,
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              accessibilityElementsHidden={true}
                              allowFontScaling={false}
                              importantForAccessibility="no-hide-descendants"
                              pointerEvents="none"
                              selectable={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#263238",
                                    "fontSize": 24,
                                  },
                                  Array [
                                    Object {
                                      "lineHeight": 24,
                                      "transform": Array [
                                        Object {
                                          "scaleX": 1,
                                        },
                                      ],
                                    },
                                    Object {
                                      "backgroundColor": "transparent",
                                    },
                                  ],
                                  Object {
                                    "fontFamily": "Material Design Icons",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              󰅖
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "justifyContent": "center",
                          "marginHorizontal": 16,
                          "marginVertical": 16,
                          "width": "100%",
                        },
                        Object {},
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "backgroundColor": "#C4C4C4",
                          "borderRadius": 180,
                          "height": 360,
                          "justifyContent": "center",
                          "marginBottom": 10,
                          "width": 360,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#E31A2C",
                              "fontSize": 170,
                            },
                            undefined,
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        󰍎
                      </Text>
                    </View>
                    <Text
                      style={
                        Object {
                          "color": "#263238",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0.2,
                          "lineHeight": 24,
                          "paddingHorizontal": 5,
                          "textAlign": "center",
                          "width": "80%",
                        }
                      }
                    >
                      Actualmente no cuentas con dirección de envío guardada.
                      <Text
                        style={
                          Array [
                            Object {
                              "fontFamily": "Roboto-Regular",
                              "fontStyle": "normal",
                              "fontWeight": "bold",
                            },
                            Object {
                              "color": "#000000",
                            },
                          ]
                        }
                      >
                         Para continuar es necesario que agregues una dirección.
                      </Text>
                    </Text>
                    <View
                      accessibilityLabel="AGREGAR DIRECCIÓN DE ENVÍO"
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#E31A2C",
                          "borderRadius": 5,
                          "flexDirection": "row",
                          "height": 40,
                          "justifyContent": "center",
                          "marginHorizontal": 16,
                          "marginTop": 12,
                          "opacity": 1,
                          "paddingVertical": 10,
                          "width": "85%",
                        }
                      }
                      testID="AGREGAR DIRECCIÓN DE ENVÍO"
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 14,
                              "fontWeight": "700",
                              "letterSpacing": 0.8,
                              "textAlign": "center",
                            },
                            undefined,
                            Object {
                              "color": "#ffffff",
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        AGREGAR DIRECCIÓN DE ENVÍO
                      </Text>
                    </View>
                  </View>
                </View>
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#f6f6f6",
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "disabled": false,
                          "expanded": false,
                        }
                      }
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "padding": 8,
                            },
                            Array [
                              Object {
                                "backgroundColor": "#ffffff",
                                "borderBottomWidth": 0.5,
                                "borderColor": "#979797",
                                "borderTopWidth": 0.5,
                                "height": 50,
                                "justifyContent": "center",
                                "width": "100%",
                              },
                            ],
                          ],
                        ]
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "margin": 8,
                              },
                              Object {
                                "flex": 1,
                                "justifyContent": "center",
                              },
                            ]
                          }
                        >
                          <Text
                            numberOfLines={1}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontWeight": "400",
                                },
                                Object {
                                  "textAlign": "left",
                                },
                                Array [
                                  Object {
                                    "fontSize": 16,
                                  },
                                  Object {
                                    "color": "rgba(0, 0, 0, 0.87)",
                                  },
                                  Object {
                                    "color": "#263238",
                                    "fontFamily": "Roboto-Regular",
                                    "fontSize": 14,
                                    "fontWeight": "600",
                                    "lineHeight": 20,
                                  },
                                ],
                              ]
                            }
                          >
                            Mis otras direcciones
                          </Text>
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "margin": 8,
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            pointerEvents="box-none"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "height": 40,
                                  "justifyContent": "center",
                                  "margin": 8,
                                  "width": 40,
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              accessibilityElementsHidden={true}
                              allowFontScaling={false}
                              importantForAccessibility="no-hide-descendants"
                              pointerEvents="none"
                              selectable={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#263238",
                                    "fontSize": 24,
                                  },
                                  Array [
                                    Object {
                                      "lineHeight": 24,
                                      "transform": Array [
                                        Object {
                                          "scaleX": 1,
                                        },
                                      ],
                                    },
                                    Object {
                                      "backgroundColor": "transparent",
                                    },
                                  ],
                                  Object {
                                    "fontFamily": "Material Design Icons",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              󰐕
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": false,
                        "expanded": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "padding": 8,
                          },
                          Array [
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderBottomWidth": 0.5,
                              "borderColor": "#979797",
                              "borderTopWidth": 0.5,
                              "height": 50,
                              "justifyContent": "center",
                              "width": "100%",
                            },
                          ],
                        ],
                      ]
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "margin": 8,
                            },
                            Object {
                              "flex": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <Text
                          numberOfLines={1}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": "#000000",
                                "fontFamily": "System",
                                "fontWeight": "400",
                              },
                              Object {
                                "textAlign": "left",
                              },
                              Array [
                                Object {
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                },
                                Object {
                                  "color": "#263238",
                                  "fontFamily": "Roboto-Regular",
                                  "fontSize": 14,
                                  "fontWeight": "600",
                                  "lineHeight": 20,
                                },
                              ],
                            ]
                          }
                        >
                          Enviar a otra dirección
                        </Text>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "margin": 8,
                            },
                            undefined,
                          ]
                        }
                      >
                        <View
                          pointerEvents="box-none"
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 40,
                                "justifyContent": "center",
                                "margin": 8,
                                "width": 40,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            accessibilityElementsHidden={true}
                            allowFontScaling={false}
                            importantForAccessibility="no-hide-descendants"
                            pointerEvents="none"
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "#263238",
                                  "fontSize": 24,
                                },
                                Array [
                                  Object {
                                    "lineHeight": 24,
                                    "transform": Array [
                                      Object {
                                        "scaleX": 1,
                                      },
                                    ],
                                  },
                                  Object {
                                    "backgroundColor": "transparent",
                                  },
                                ],
                                Object {
                                  "fontFamily": "Material Design Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            󰐕
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": false,
                        "expanded": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "padding": 8,
                          },
                          Array [
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderBottomWidth": 0.5,
                              "borderColor": "#979797",
                              "borderTopWidth": 0.5,
                              "height": 50,
                              "justifyContent": "center",
                              "width": "100%",
                            },
                          ],
                        ],
                      ]
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "margin": 8,
                            },
                            Object {
                              "flex": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <Text
                          numberOfLines={1}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": "#000000",
                                "fontFamily": "System",
                                "fontWeight": "400",
                              },
                              Object {
                                "textAlign": "left",
                              },
                              Array [
                                Object {
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                },
                                Object {
                                  "color": "#263238",
                                  "fontFamily": "Roboto-Regular",
                                  "fontSize": 14,
                                  "fontWeight": "600",
                                  "lineHeight": 20,
                                },
                              ],
                            ]
                          }
                        >
                          Artículos envío a domicilio
                        </Text>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "margin": 8,
                            },
                            undefined,
                          ]
                        }
                      >
                        <View
                          pointerEvents="box-none"
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 40,
                                "justifyContent": "center",
                                "margin": 8,
                                "width": 40,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            accessibilityElementsHidden={true}
                            allowFontScaling={false}
                            importantForAccessibility="no-hide-descendants"
                            pointerEvents="none"
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "#263238",
                                  "fontSize": 24,
                                },
                                Array [
                                  Object {
                                    "lineHeight": 24,
                                    "transform": Array [
                                      Object {
                                        "scaleX": 1,
                                      },
                                    ],
                                  },
                                  Object {
                                    "backgroundColor": "transparent",
                                  },
                                ],
                                Object {
                                  "fontFamily": "Material Design Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            󰐕
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
          <Modal
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          />
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
